include/bactria/Event.hpp

```diff
@@ -13,62 +13,180 @@
+                std::swap(m_handle, other.m_handle);
+            }
 
+            /**
+             * \brief The move assignment operator.
+             *
+             * The move assignment operator will move the contents of the original Event into this Event. The original
+             * Event will be in an invalid state; using it is undefined behaviour.
+             *
+             * \param rhs The Event to be moved.
+             */
+            auto operator=(Event&& rhs) noexcept -> Event&
+            {
+                Marker::operator=(std::forward<Event>(rhs));
+                std::swap(m_handle, rhs.m_handle);
+                m_action = std::move(rhs.m_action);
+
+                return *this;
+            }
+
+            /**
+             * \brief The destructor.
+             *
+             * The destructor will discard all internal contents of the Event. After the call to the destructor the
+             * Event will be in an invalid state; using it is undefined behaviour.
+             */
             ~Event() override
             {
-                plugin::destroy_event(m_handle);
+                if(detail::is_activated())
+                    plugin::destroy_event(m_handle);
             }
 
-            auto fire() noexcept -> void
+            /**
+             * \brief Fire the event.
+             *
+             * Firing the event generates an entry on the visualizer with this event's name, color and category.
+             * While the interface requires the source file, the line number and the calling function, this
+             * information may not be supported by all back-ends. In this case the parameters will be silently
+             * ignored.
+             * 
+             * \param source The source file where the event is fired. Should be `__FILE__`.
+             * \param lineno The source line where the event is fired. Should be `__LINE__`.
+             * \param caller The surrounding function of the event firing. Should be `__func__`.
+             */
+            auto fire(std::string source, std::uint32_t lineno, std::string caller) noexcept -> void
             {
-                plugin::fire_event(m_handle);
+                if(detail::is_activated())
+                {
+                    plugin::fire_event(m_handle, m_action().c_str(), source.c_str(), lineno, caller.c_str());
+                }
+            }
+
+            /**
+             * \brief Set a user-defined action for generating the event name.
+             *
+             * This method allows to specify a user-defined action which generates the name. It will be executed
+             * upon firing the event. This method will override the name specified in the constructor.
+             *
+             * \param a The action to execute upon firing the event.
+             * \sa fire
+             */
+            auto set_action(std::function<std::string(void)> a) noexcept -> void
+            {
+                m_action = a;
```

Comment:
`m_action = std::move(a)`

Otherwise, the copy assignment could throw, which violates your exception specification :)